contracts/RecoveryQuorum.sol

```diff
@@ -3,111 +3,138 @@ import "./RecoverableController.sol";
 import "./libs/ArrayLib.sol";
 
 contract RecoveryQuorum {
-  RecoverableController public controller;
+    RecoverableController public controller;
+    uint constant MAX_DELEGATES = 15;
+    uint constant NEVER = 31536000000000; // large number that acts as inf
 
-  address[] public delegateAddresses; // needed for iteration of mapping
-  mapping (address => Delegate) public delegates;
-  struct Delegate{
-    uint deletedAfter; // delegate exists if not 0
-    uint pendingUntil;
-    address proposedUserKey;
-  }
+    address[] public delegateAddresses; // needed for iteration of mapping
+    mapping (address => Delegate) public delegates;
+    struct Delegate {
+        uint deletedAfter; // delegate exists if not 0
+        uint pendingUntil;
+        address proposedUserKey;
+    }
+
+    event RecoveryEvent(string action, address initiatedBy);
+
+    modifier onlyUserKey(){ if (msg.sender == controller.userKey()) _; }
 
-  event RecoveryEvent(string action, address initiatedBy);
+    function RecoveryQuorum(address _controller, address[] _delegates) {
+        controller = RecoverableController(_controller);
+        for(uint i = 0; i < _delegates.length; i++) {
+            if (i >= MAX_DELEGATES) {
```

Comment:
Yes, that makes sense. I haven't put that much effort into updating the RecoveryQuorum since I don't know how long we are planning on using it.